shell/src/main/java/alluxio/cli/bundler/command/CollectLogCommand.java

```diff
@@ -14,21 +14,122 @@
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * Command to collect Alluxio logs.
  * */
 public class CollectLogCommand  extends AbstractCollectInfoCommand {
   public static final String COMMAND_NAME = "collectLog";
   private static final Logger LOG = LoggerFactory.getLogger(CollectLogCommand.class);
+  public static final Set<String> FILE_NAMES = Stream.of(
+      "master.log",
+      "master.out",
+      "job_master.log",
+      "job_master.out",
+      "master_audit.log",
+      "worker.log",
+      "worker.out",
+      "job_worker.log",
+      "job_worker.out",
+      "proxy.log",
+      "proxy.out",
+      "task.log",
+      "task.out",
+      "user"
+  ).collect(Collectors.toSet());
+  // We tolerate the beginning of a log file to contain some rows that are not timestamped.
+  // 30 is chosen because a YARN application log can have >20 rows in the beginning for
+  // general information about a job.
+  // The timestamped log entries start after this general information block.
+  private static final int TRY_PARSE_LOG_ROWS = 30;
+
+  // Preserves the order of iteration, we try the longer pattern before the shorter one
+  private static final Map<DateTimeFormatter, Integer> FORMATTERS =
+          new LinkedHashMap<DateTimeFormatter, Integer>(){
+    {
+      // "2020-01-03 12:10:11,874"
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss,SSS"), 23);
+      // "2020-01-03 12:10:11"
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"), 19);
+      // "2020-01-03 12:10"
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"), 16);
+      // "20/01/03 12:10:11"
+      put(DateTimeFormatter.ofPattern("yy/MM/dd HH:mm:ss"), 17);
+      // "20/01/03 12:10"
+      put(DateTimeFormatter.ofPattern("yy/MM/dd HH:mm"), 14);
+      // 2020-01-03T12:10:11.874+0800
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXX"), 28);
+      // 2020-01-03T12:10:11
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss"), 19);
+      // 2020-01-03T12:10
+      put(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm"), 16);
+    }
+  };
+
+  private String mLogDirPath;
+  private File mLogDir;
+  private URI mLogDirUri;
+  private Set<String> mIncludedPrefix;
+  private Set<String> mExcludedPrefix;
+  private LocalDateTime mStartTime;
+  private LocalDateTime mEndTime;
+
+  public static final String INCLUDE_OPTION_NAME = "include-logs";
+  private static final Option INCLUDE_OPTION =
+          Option.builder().required(false).argName("filename-prefixes")
+                  .longOpt(INCLUDE_OPTION_NAME).hasArg(true)
+                  .desc("extra log file name prefixes to include in ${ALLUXIO_HOME}/logs. "
```

Comment:
what is the order of operation between include and exclude? For example, if `master.log` is in both include and exclude, what happens?

Reply:
The definition of this `--include-logs` option was not very clear. Now it's changed to the below 3 options:

`--include-logs`: replace the filename prefixes with what is defined here. Does not work tgt with the other two options (err out if given together).

`--exclude-logs`: minus from the default filenames. This is checked before `--additional-logs`. If "master.log" is in both `exclude` and `additional`, it will be excluded. This caveat is now in the option description.

`--additional-logs`: add to the default filenames. Checked after the exclusion.